src/errors.rs

```diff
@@ -0,0 +1,72 @@
+use core::fmt::{Display, Formatter, Result};
+
+use derive_more::Error;
+
+#[cfg_attr(feature = "std", derive(Error))]
+#[derive(Debug, PartialEq)]
+pub enum ArithmeticError {
+    Overflow,
+    DivisionByZero,
+}
+
+impl ArithmeticError {
+    pub const fn as_str(&self) -> &'static str {
+        use ArithmeticError::*;
+        match self {
+            Overflow => "overflow",
```

Comment:
jfyi: u can use `Self::Overflow` instead of extra `use`

Reply:
fixed